flink-runtime/src/test/java/org/apache/flink/runtime/rpc/akka/AkkaRpcActorTest.java

```diff
@@ -560,6 +561,47 @@ public void canRespondWithSerializedValueLocally() throws Exception {
         }
     }
 
+    /**
+     * Verifies that actions scheduled via the main thread executor are eventually run while
+     * adhering to the provided delays.
+     *
+     * <p>This test does not assert any upper bounds for how late something is run, because that
+     * would make the test unstable in some environments, and there is no guarantee that such an
+     * upper bound exists in the first place.
+     *
+     * <p>There are various failure points for this test, including the scheduling from the {@link
+     * RpcEndpoint} to the {@link AkkaInvocationHandler}, the conversion of these calls by the
+     * handler into Call-/RunAsync messages, the handling of said messages by the {@link
+     * AkkaRpcActor} and in the case of RunAsync the actual scheduling by the underlying actor
+     * system. This isn't an ideal test setup, but these components are difficult to test in
+     * isolation.
+     */
+    @Test
+    public void testScheduling() throws ExecutionException, InterruptedException {
```

Comment:
yes this test could be split into 3 separate test cases but it seems too mundane to warrant the additional effort.